### ISSUE TITLE
Release build compiler warning

### ISSUE THREAD

**Issue by Myoldmopar** [NREL/EnergyPlus#7119]
Issue overview
--------------
I only noticed them during a release build.  I believe the debug build is clean now.

```
/eplus/repos/1eplus/src/EnergyPlus/WeatherManager.cc: In function ‘void EnergyPlus::WeatherManager::SetCurrentWeather()’:
/eplus/repos/1eplus/src/EnergyPlus/WeatherManager.cc:2490:34: warning: ‘%02d’ directive writing between 2 and 11 bytes into a region of size between 0 and 4 [-Wformat-overflow=]
         std::sprintf(time_stamp, "%02d/%02d %02d:", Month, DayOfMonth, HourOfDay - 1);
                                  ^~~~~~~~~~~~~~~~~
/eplus/repos/1eplus/src/EnergyPlus/WeatherManager.cc:2490:34: note: directive argument in the range [-2147483648, 2147483646]
In file included from /usr/include/stdio.h:873,
                 from /usr/include/c++/8/cstdio:42,
                 from /usr/include/c++/8/ext/string_conversions.h:43,
                 from /usr/include/c++/8/bits/basic_string.h:6391,
                 from /usr/include/c++/8/string:52,
                 from /usr/include/c++/8/stdexcept:39,
                 from /usr/include/c++/8/array:39,
                 from /eplus/repos/1eplus/src/EnergyPlus/WeatherManager.cc:49:
```

```
In file included from /eplus/repos/5eplus/third_party/ObjexxFCL/src/ObjexxFCL/gio_Fmt.hh:17,
                 from /eplus/repos/5eplus/src/EnergyPlus/DataSizing.hh:54,
                 from /eplus/repos/5eplus/tst/EnergyPlus/unit/VAVDefMinMaxFlow.unit.cc:56:
/eplus/repos/5eplus/third_party/ObjexxFCL/src/ObjexxFCL/Format.hh: In constructor ‘ObjexxFCL::Format::Format(ObjexxFCL::Format*, char)’:
/eplus/repos/5eplus/third_party/ObjexxFCL/src/ObjexxFCL/Format.hh:126:35: warning: type qualifiers ignored on cast result type [-Wignored-qualifiers]
   static_cast< void const >( star ); // Suppress unused warning
                                   ^
```

### Details
Some additional details for this issue (if relevant):
 - Platform: Ubuntu 18.10 (GCC 8.2.0)

**Comment by Myoldmopar** [comment]
Partially fixed with #7120 but not entirely.  Remaining build warnings in release build for me:

```
/eplus/repos/4eplus/src/ReadVars/ReadVarsESO.f90:419:0:

               line=line(2:pos-1)
 
Warning: ‘__builtin_memmove’ reading 500 bytes from a region of size 499 [-Wstringop-overflow=]
/eplus/repos/4eplus/src/ReadVars/ReadVarsESO.f90:662:0:

                 stovar(nstore)=line(1:i-1)
 
Warning: ‘__builtin_memmove’ reading 1500 bytes from a region of size 500 [-Wstringop-overflow=]
/eplus/repos/4eplus/src/ReadVars/ReadVarsESO.f90:707:0:

               trackvar(ij)=line(1:i-1)
 
Warning: ‘__builtin_memmove’ reading 1500 bytes from a region of size 500 [-Wstringop-overflow=]
/eplus/repos/4eplus/src/ReadVars/ReadVarsESO.f90:793:0:

               trackvar(ij)=line(1:i-1)
 
Warning: ‘__builtin_memmove’ reading 1500 bytes from a region of size 500 [-Wstringop-overflow=]
```

```
In file included from /eplus/repos/4eplus/third_party/gtest/googletest/include/gtest/gtest.h:58,
                 from /eplus/repos/4eplus/third_party/ObjexxFCL/tst/unit/Required.unit.cc:14:
/eplus/repos/4eplus/third_party/ObjexxFCL/tst/unit/Required.unit.cc: In member function ‘virtual void RequiredTest_ConstructionOmit_Test::TestBody()’:
/eplus/repos/4eplus/third_party/ObjexxFCL/tst/unit/Required.unit.cc:35:40: warning: unnecessary parentheses in declaration of ‘_’ [-Wparentheses]
  EXPECT_DEBUG_DEATH( Required_int_const( _ ), ".*Assertion.*" );
                                        ^
/eplus/repos/4eplus/third_party/gtest/googletest/include/gtest/internal/gtest-internal.h:1122:44: note: in definition of macro ‘GTEST_SUPPRESS_UNREACHABLE_CODE_WARNING_BELOW_’
   if (::testing::internal::AlwaysTrue()) { statement; }
                                            ^~~~~~~~~
/eplus/repos/4eplus/third_party/gtest/googletest/include/gtest/gtest-death-test.h:259:3: note: in expansion of macro ‘GTEST_EXECUTE_STATEMENT_’
   GTEST_EXECUTE_STATEMENT_(statement, regex)
   ^~~~~~~~~~~~~~~~~~~~~~~~
/eplus/repos/4eplus/third_party/ObjexxFCL/tst/unit/Required.unit.cc:35:2: note: in expansion of macro ‘EXPECT_DEBUG_DEATH’
  EXPECT_DEBUG_DEATH( Required_int_const( _ ), ".*Assertion.*" );
  ^~~~~~~~~~~~~~~~~~
[ 50%] Building CXX object third_party/ObjexxFCL/CMakeFiles/objexx_tests.dir/tst/unit/Stream.unit.cc.o
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-vexing-parse’
```

```
/eplus/repos/4eplus/src/EnergyPlus/WeatherManager.cc: In function ‘void EnergyPlus::WeatherManager::SetCurrentWeather()’:
/eplus/repos/4eplus/src/EnergyPlus/WeatherManager.cc:2490:34: warning: ‘%02d’ directive writing between 2 and 11 bytes into a region of size between 0 and 4 [-Wformat-overflow=]
         std::sprintf(time_stamp, "%02d/%02d %02d:", Month, DayOfMonth, HourOfDay - 1);
                                  ^~~~~~~~~~~~~~~~~
/eplus/repos/4eplus/src/EnergyPlus/WeatherManager.cc:2490:34: note: directive argument in the range [-2147483648, 2147483646]
In file included from /usr/include/stdio.h:873,
                 from /usr/include/c++/8/cstdio:42,
                 from /usr/include/c++/8/ext/string_conversions.h:43,
                 from /usr/include/c++/8/bits/basic_string.h:6391,
                 from /usr/include/c++/8/string:52,
                 from /usr/include/c++/8/stdexcept:39,
                 from /usr/include/c++/8/array:39,
                 from /eplus/repos/4eplus/src/EnergyPlus/WeatherManager.cc:49:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:36:34: note: ‘__builtin___sprintf_chk’ output between 10 and 37 bytes into a destination of size 10
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by Myoldmopar** [comment]
The weather one is the only one that remains, opening up a specific issue for it.
